ext/standard/array.c

```diff
@@ -3228,7 +3228,7 @@ PHP_FUNCTION(array_pop)
 	}
 	ZVAL_COPY_DEREF(return_value, val);
 
-	if (!p->key && Z_ARRVAL_P(stack)->nNextFreeElement > 0 && p->h >= (zend_ulong)(Z_ARRVAL_P(stack)->nNextFreeElement - 1)) {
+	if (!p->key && (zend_long)p->h >= (Z_ARRVAL_P(stack)->nNextFreeElement - 1)) {
```

Comment:
This still doesn't look right to me. If you have something like

```
$a = [-2 => 'a', -1 => 'b', 0 => 'c'];
array_pop($a);
array_pop($a);
$a[] = 'd';
```

the end result should be `[-2 => 'a', -1 => 'd']`, which I don't think is the case with this implementation.

Reply:
With the current patch that's the result I get. Can you explain a bit better what looks wrong or have another example?

Reply:
Actually, I think this code is just written in a very non-obvious manner, and this should really just be `p->h == (Z_ARRVAL_P(stack)->nNextFreeElement - 1)`. Does anything break if you change it to that?

Reply:
Indeed, I can't think of a case where the `>` would make sense. `make test` came back green so I pushed that change